### DISCUSSION TITLE
smart_union removed, is it by default in v2?

### DISCUSSION THREAD

**Discussion by deivydaslipskis** [pydantic/pydantic#7264] [Question] [answered]
Hello, I have noticed that `smart_union` has been removed.

Will Pydantic v2 check all allowed types before even trying to coerce by default? As it was available in v1 - https://docs.pydantic.dev/1.10/usage/model_config/#smart-union

Or do I need to set smart_union in another way? In the Migration guide, I could not find any additional information about `smart_union`.

**Comment by PrettyWood** [comment, marked as answer]
By default, v2 has the "smart union" behaviour. If you want to keep the behaviour of v1 you can use
```py
id: int | str = Field(..., union_mode='left_to_right')
```

**Reply by deivydaslipskis** [reply]
Thanks! I am migrating to v2 and was wondering how to use it.
